### ISSUE TITLE
Emergency Vehicle

### ISSUE THREAD

**Issue by CarloVanDriestenBMW** [OpenSimulationInterface/open-simulation-interface#294]
We should add the possibility of indicating if an emergency vehicle is currently "on duty".

[Missing vehicle type?](https://opensimulationinterface.github.io/open-simulation-interface/structosi3_1_1MovingObject_1_1VehicleClassification.html)

**Comment by mbencik999** [comment]
Is the idea to flag the vehicle with a enum type or crate a new message? If the emergency vehicles have new properties a new message type could be considered.

**Comment by CarloVanDriestenBMW** [comment]
Yes, both options - I am actually not really happy with the way the moving object and the vehicle is constructed - maybe we have the possibility to restructure the message in 4.0.0. Let's see how many other issues connected with the vehicle type we can collect in the next time.

**Comment by CezarySzyszkaAltran** [comment]
Maybe temporary solution would be to add to osi_object.proto MovingObject.VehicleAttributes.VehicleClassification.LightState something linke

optional GenericLightState privileged_vehicle_ilumination = 8; [For vehicles that do not have to follow traffic rules all the time]

optional GenericLightState  special_vehicle_ilumination = 9; [For vehicles that can behave erratically like oversize trucks, garbage truck, snow removal, etc. ]
And in the same file
enum GenericLightState
      GENERIC_LIGHT_STATE_FLASHING_BLUE = 4;
      GENERIC_LIGHT_STATE_FLASHING_BLUE_AND_RED = 4;
      GENERIC_LIGHT_STATE_FLASHING_AMBER = 5;

This would allow us to cover not only emergency vehicles but also special vehicles and does not break anything.

**Comment by CarloVanDriestenBMW** [comment]
I like the idea!

**Comment by EngelhardtNickBMW** [comment]
I think the proposed LightState solves the immediate issue nicely.

 Although for future simulations, I think a special vehicle type for all vehicles that can have special rights could be justified. An example that comes to mind could be police vehicles signalling our ego-vehicle to stop/pull over/follow.

**Comment by CarloVanDriestenBMW** [comment]
OK, so we leave this issue as a reminder. @CezarySzyszkaAltran  could you propose a Pullrequest with you change, please? I would like to merge it right away for the release on friday!
